Release 5.0.9 (#3253)
ezsystems/developer-documentation: docs/release_notes/ibexa_dxp_v5.0.md

<div class="release-notes" markdown="1">

[[% set version = 'v5.0.9' %]]
[[% set date = '2026-06-30' %]]

[[= release_note_entry_begin(
    'MCP Servers ' + version,
    date,
    ['Headless', 'Experience', 'Commerce', 'LTS Update', 'New feature', 'First release']
) =]]

### Tools

> Several new content type tools were added to the MCP Servers feature:

Reviewer: ```suggestion
Several new experimental content type tools were added to the MCP Servers feature:
```

- `create_content_type`
- `get_content_type`
- `get_content_type_by_identifier`
- `get_content_type_list`
- `get_content_type_draft`
- `publish_content_type_draft`
- `add_field_definition`
- `remove_field_definition`
- `update_field_definition`
- `get_content_type_groups`

Among translation tools:

- `list_non_translated_content_ids` tool is added.
- `list_content_translations` is now renamed `list_content_languages`

For more information, see [Built-in tools](https://doc.ibexa.co/en/5.0/ai/mcp/mcp_config/#built-in-tools).

### Configuration

> - A mandatory `allowed_hosts` parameter is added to configuration to restrict access to an MCP server. For more information, see [Allowed hosts](https://doc.ibexa.co/en/5.0/ai/mcp/mcp_config/#allowed-hosts).

Reviewer: It's not truly mandatory, it's just that the default value covers so little, that this is almost mandatory to set a value. I could be more strict and honest:

```suggestion
- An `allowed_hosts` parameter is added to configuration to restrict access to an MCP server. It's default value covers only few local cases. For more information, see [Allowed hosts](https://doc.ibexa.co/en/5.0/ai/mcp/mcp_config/#allowed-hosts).
```

<div class="release-notes" markdown="1">

[[% set version = 'v5.0.9' %]]
[[% set date = '2026-06-30' %]]

[[= release_note_entry_begin(
    'MCP Servers ' + version,
    date,
    ['Headless', 'Experience', 'Commerce', 'LTS Update', 'New feature', 'First release']
) =]]

### Tools

Several new content type tools were added to the MCP Servers feature:

- `add_field_definition`
- `create_content_type`
- `get_content_type_by_identifier`
- `get_content_type_draft`
- `get_content_type_groups`
- `get_content_type_list`
- `get_content_type`
- `list_non_translated_content_ids`
- `publish_content_type_draft`
- `remove_field_definition`

> - `update_field_definition`

Reviewer: ```suggestion
Several new content type tools are available.
```

<div class="release-notes" markdown="1">

[[% set version = 'v5.0.9' %]]
[[% set date = '2026-06-30' %]]

[[= release_note_entry_begin(
    'MCP Servers ' + version,
    date,

>     ['Headless', 'Experience', 'Commerce', 'LTS Update', 'New feature', 'First release']

Reviewer 2: ```suggestion
    ['Headless', 'Experience', 'Commerce', 'LTS Update', 'New feature']
```
MCP was released already

) =]]

### Tools

Several new experimental content type tools were added to the MCP Servers feature:

- `create_content_type`
- `get_content_type`
- `get_content_type_by_identifier`
- `get_content_type_list`
- `get_content_type_draft`
- `publish_content_type_draft`
- `add_field_definition`
- `remove_field_definition`
- `update_field_definition`
- `get_content_type_groups`

Among translation tools:

- `list_non_translated_content_ids` tool is added.

> - `list_content_translations` is now renamed `list_content_languages`

Reviewer: From https://github.com/ibexa/documentation-developer/pull/3248#discussion_r3491237644
```suggestion
- `list_content_translations` is now renamed to `list_content_languages`
```

For more information, see [Built-in tools](https://doc.ibexa.co/en/5.0/ai/mcp/mcp_config/#built-in-tools).

### Configuration

> - An `allowed_hosts` parameter is added to configuration to restrict access to an MCP server. It's default value covers only few local cases. For more information, see [Allowed hosts](https://doc.ibexa.co/en/5.0/ai/mcp/mcp_config/#allowed-hosts).

Reviewer 2: ```suggestion
- An `allowed_hosts` parameter is added to configuration to restrict access to an MCP server. It's default value covers only few cases for local development. For more information, see [Allowed hosts](https://doc.ibexa.co/en/5.0/ai/mcp/mcp_config/#allowed-hosts).
```
"local cases" isn't clear enough imho

> - A `title` property is added to capability attributes to optionally provide a friendly UI label. For more information see [MCP server capabilities](https://doc.ibexa.co/en/5.0/ai/mcp/mcp_usage/#mcp-server-capabilities).

Reviewer 2: ```suggestion
- A `title` property is added to capability attributes to optionally provide a friendly UI label. For more information, see [MCP server capabilities](https://doc.ibexa.co/en/5.0/ai/mcp/mcp_usage/#mcp-server-capabilities).
```

[[= release_note_entry_end() =]]

[[= release_note_entry_begin(
    "Ibexa DXP " + version,

>     '2026-06-30',

Reviewer 2: ```suggestion
    date,
```
To make sure we only change one place in case the release is postponed.

    ['Headless', 'Experience', 'Commerce', 'New feature']
) =]]

### Hybrid tracking

New `hybrid` tracking mode is available alongside [`client` and `server`](tracking_functions.md).
In this mode, the browser uses a first-party tracking shim provided by the DXP instance.
Tracking events are forwarded through a same-origin endpoint and processed server side before being sent to Raptor, helping reduce the impact of ad blockers while preserving client side event tracking.

For more information, see [hybrid tracking](https://doc.ibexa.co/en/5.0/recommendations/raptor_integration/hybrid_tracking/).

> ### Raptor connector

Reviewer 2: Shouldn't Raptor connector be the main header?

```
### Raptor connector

#### Hybrid tracking

#### New recommendation blocks
```

Seems unusual to have Hybrid tracking outside the "Raptor connector" section

> <div class="release-notes" markdown="1">

Reviewer 2: We need to copy the 4.6 release notes to this PR as well

[[% set version = 'v5.0.9' %]]

> [[% set date = '2026-06-30' %]]

Reviewer 2: ```suggestion
[[% set date = '2026-07-01' %]]
```
